src/Codeception/Command/Shared/FileSystem.php

```diff
@@ -60,12 +60,26 @@ protected function removeSuffix($classname, $suffix)
         return preg_replace("~$suffix$~", '', $classname);
     }
 
-    protected function save($filename, $contents, $force = false, $flags = null)
+    protected function save($filename, $contents, $force = false)
     {
         if (file_exists($filename) && !$force) {
             return false;
         }
-        file_put_contents($filename, $contents, $flags);
+
+        //Write in a locked temporary file in order to reduce non thread-safe disk instructions,
+        //  and reduce risk of `Bus Errors` when running tests in parallel.
+        //@see https://github.com/Codeception/Codeception/issues/2054#event-1046859271
+        $tmpFilename = $filename . '.tmp';
```

Comment:
I would've used [tempnam](https://secure.php.net/manual/en/function.tempnam.php) instead of concatenating the constant `.tmp`.